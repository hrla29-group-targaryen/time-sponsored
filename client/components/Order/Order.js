import React from 'react'

import Delivery from './Delivery/Delivery'
import Pickup from './Pickup/Pickup'
import Schedule from './Schedule/Schedule'

import './Order.css'

class Order extends React.Component{
	constructor(props){
		super(props)
		this.state={
			route : 'delivery'
		}
		this.route_changer=this.route_changer.bind(this)
		this.view_controller=this.view_controller.bind(this)
	}

	route_changer(route){
		this.setState({route})
	}

	view_controller(route){
		if (route === 'delivery') {
			return <Delivery toggleModal={this.props.toggleModal} route_changer={this.route_changer}/>
		} else if (route === 'pickup') {
			return <Pickup toggleModal={this.props.toggleModal} route_changer={this.route_changer}/>
		} else if (route === 'schedule') {
			return <Schedule toggleModal={this.props.toggleModal} route_changer={this.route_changer}/>
 		}
	}

	render(){
		return(
			<section className="order-modal"> 
				{
					this.view_controller(this.state.route)
				}
			</section>
			)
	}
}

export default Order


			
			// <section className="order-card slide-in-bottom">
			// 	<section id="order-header">
			// 		<button onClick={toggleModal}>close</button>
			// 	</section>
			// 	<section>
			// 		<h1>Your order settings</h1>
			// 	</section>
			// 	<section className="order-put-it-center order-button1" >
			// 		<button>Delivery</button>
			// 		<button>Pickup</button>
			// 	</section>
			// 	<section>
			// 		When would you like to order?
			// 	</section>

			// </section>
