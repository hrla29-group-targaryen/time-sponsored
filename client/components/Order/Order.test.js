import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Order from './Order';

describe('testing Order component', ()=> {

	// it('should grab a snapshot of the component.', ()=> {
	// 	const component = renderer.create(<Order />)
	// 	let tree = component.toJSON();
	// 	expect(tree).toMatchSnapshot();
	// })

	it('should return a single-node wrapper.', ()=> {
		expect(shallow(<Order />).length).toEqual(1)
	})

	it('should check state ', ()=> {
		let wrapper = shallow(<Order />)
		expect.assertions(1)
		expect(wrapper.state().route).toEqual('delivery')	
	})

	it('should check state ', ()=> {
		let wrapper = shallow(<Order />)
		expect.assertions(1)
		wrapper.instance().route_changer("pickup")
		expect(wrapper.state().route).toEqual('pickup')		
	})

	it('should check state ', ()=> {
		let wrapper = shallow(<Order />)
		expect.assertions(3)
		expect(wrapper.instance().view_controller("delivery")).toBeTruthy()	
		expect(wrapper.instance().view_controller("pickup")).toBeTruthy()	
		expect(wrapper.instance().view_controller("schedule")).toBeTruthy()		
	})
})
