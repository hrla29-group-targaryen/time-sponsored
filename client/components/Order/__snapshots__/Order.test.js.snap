// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Order component should grab a snapshot of the component. 1`] = `
<section
  className="order-modal"
>
  <section
    className="order-delivery-container"
  >
    <section
      className="order-delivery-exit"
    >
      <button>
        X
      </button>
      <h3>
        Your order settings
      </h3>
    </section>
    <section
      className="order-delivery-pickup-buttons"
    >
      <button>
        Delivery
      </button>
      <button
        onClick={[Function]}
      >
        Pickup
      </button>
    </section>
    <section
      className="order-delivery-whenwouldyou"
    >
      <h5>
        When would you like your order?
      </h5>
      <button
        onClick={[Function]}
      >
        ASAP (35â€“45m)
      </button>
    </section>
    <section
      className="order-delivery-address"
    >
      <h5>
        Delivery address
      </h5>
      <div
        className="order-delivery-input-container"
      >
        <i
          className="fas fa-map-marker-alt"
        />
        <input
          className="order-delivery-input-field"
          placeholder="Street address, city, state"
        />
      </div>
    </section>
    <section
      className="order-delivery-update"
    >
      <button>
        Update
      </button>
    </section>
  </section>
</section>
`;
